docs/docs/native-docker-private-registry.md

```diff
@@ -4,34 +4,176 @@ title: Using a Private Docker Registry
 
 # Using a Private Docker Registry
 
-This document describes how to initiate a [Docker](https://docker.com/) pull from
-an authenticated private registry.
+As of Marathon 1.5, you can upload your private Docker registry credentials to a secret store, then reference them in your app or pod definition. This functionality is only available if you are using the Mesos containerizer. If you are using the Docker containerizer, follow [these instructions](#docker-containerizer) to use a private Docker registry.
 
-## Registry  1.0 - Docker pre 1.6 
+## Step 1: Create a Credentials File
+
+1. Log in to your private registry manually. This will create a `.docker` folder and a `.docker/config.json` file.
+  ```bash
+  $ docker login some.docker.host.com
+  Username: foo
+  Password:
+  Email: foo@bar.com
+  ```
+
+1. Check that you have the `.docker/config.json` file.
+
+    ```bash
+    $ ls .docker
+    config.json
+    ```
+
+    Your `config.json` file should look like this.
+
+    ```json
+    {
+      "auths": {
+          "https://index.docker.io/v1/": {
+              "auth": "aWNoZXJuZXRza3k6YlIoKW5TME4jZG8="
+          }
+      }
+    }
+    ```
+
+1. Add the `config.json` file to a secret store. If you are using Enterprise DC/OS, [follow these instructions to add the file to the DC/OS secret store](https://docs.mesosphere.com/1.9/security/secrets/create-secrets/#api).
+  **Note:** As of DC/OS version 10.0, you can only add a file to the secret store via the API or the DC/OS CLI.
+
+### Step 2: Add the Secret to your App or Pod Definition
+
+#### For an Application
+
+Add the following two parameters to your app definition.
+
+1. A location for the secret in the `secrets` parameter:
+    ```json
+    "secrets": {
+          "pullConfigSecret": {
+              "source": "/mesos-docker/pullConfig"
+          }
+      }
+    ```
+
+1. A reference to the secret in the `docker.pullConfig` parameter:
+    ```json
+    "docker": {
+            "image": "mesosphere/inky",
+            "pullConfig": {
+                "secret": "pullConfigSecret"
+              }
+    }
+    ```
+
+  **Note:** This functionality is _only_ supported with the Mesos containerizer: `container.type` must be `MESOS`.
+
+1. A complete example:
+
+      ```json
+      {
```

Comment:
The indentation is consistent within the example, and not consistent throughout the document. 2 vs 4 vs N space characters.